packages/core/src/components/ui/ProductGallery/ProductGallery.tsx

```diff
@@ -212,14 +216,15 @@ function ProductGallery({
             </div>
           )}
           {/* Render ALL products */}
-          {productGalleryData ? (
+          {data?.search?.facets.length > 0 ? (
```

Comment:
`facets` can be undefined and `facets.length` will return an error, no?

Reply:
yeap, although I think this is not easy to happen hahaha. Nice catch, ty!
I change this piece of [code here](https://github.com/vtex/faststore/pull/1967/commits/4562407ec3ec7dfb9050d257643ddaa21016ea23), also extracting to a new const.